File: src/types/api/assistant.ts
/**
 * Assistant API types - matching backend structure
 */

export interface Assistant {
  id: string
  name: string
  description?: string
  instructions?: string
  parameters?: Record<string, any>
  created_by?: string
  is_template: boolean
  is_active: boolean
  created_at: string
  updated_at: string
}

export interface CreateAssistantRequest {
  name: string
  description?: string
  instructions?: string
  parameters?: Record<string, any>
  is_template?: boolean
}

export interface UpdateAssistantRequest {
  name?: string
  description?: string
  instructions?: string
  parameters?: Record<string, any>
  is_template?: boolean
  is_active?: boolean
}

export interface AssistantListResponse {
  assistants: Assistant[]
  total: number
  page: number
  per_page: number
}

// Default assistant parameters
export const DEFAULT_ASSISTANT_PARAMETERS = {
  stream: true,
  temperature: 0.7,
  frequency_penalty: 0.7,
  presence_penalty: 0.7,
  top_p: 0.95,
  top_k: 2,
} as const

File: src-tauri/src/database/models/assistant.rs
use super::model::ModelParameters;
use crate::database::types::JsonOption;
use chrono::{DateTime, Utc};
use schemars::JsonSchema;
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
pub struct Assistant {
    pub id: Uuid,
    pub name: String,
    pub description: Option<String>,
    pub instructions: Option<String>,
    pub parameters: JsonOption<ModelParameters>,
    pub created_by: Option<Uuid>,
    pub is_template: bool,
    pub is_default: bool,
    pub is_active: bool,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

// Request/Response structures for assistants
#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
pub struct CreateAssistantRequest {
    pub name: String,
    pub description: Option<String>,
    pub instructions: Option<String>,
    pub parameters: Option<ModelParameters>,
    pub is_template: Option<bool>,
    pub is_default: Option<bool>,
}

#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
pub struct UpdateAssistantRequest {
    pub name: Option<String>,
    pub description: Option<String>,
    pub instructions: Option<String>,
    pub parameters: Option<ModelParameters>,
    pub is_template: Option<bool>,
    pub is_default: Option<bool>,
    pub is_active: Option<bool>,
}

#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
pub struct AssistantListResponse {
    pub assistants: Vec<Assistant>,
    pub total: i64,
    pub page: i32,
    pub per_page: i32,
}
